go/test/endtoend/cluster/vttablet_process.go

```diff
@@ -143,6 +153,25 @@ func (vttablet *VttabletProcess) WaitForStatus(status string) bool {
 	return false
 }
 
+// GetTabletStatus function checks if vttablet process is up and running
+func (vttablet *VttabletProcess) GetTabletStatus() string {
+	resp, err := http.Get(vttablet.VerifyURL)
+	if err != nil {
+		return ""
+	}
+	if resp.StatusCode == 200 {
+		resultMap := make(map[string]interface{})
+		respByte, _ := ioutil.ReadAll(resp.Body)
+		err := json.Unmarshal(respByte, &resultMap)
+		if err != nil {
+			panic(err)
+		}
+		status := reflect.ValueOf(resultMap["TabletStateName"]).String()
+		return status
+	}
```

Comment:
There is already similar code in `WaitForStatus` may be we can merge this and check for `NOT_SERVING` status in that function.

Reply:
done